### DISCUSSION TITLE
Is it possible for an island to interact with another?

### DISCUSSION THREAD

**Discussion by italodeandra** [freshframework/fresh#441] [Q&A] [answered]
Quoting from the documentation:

> "islands" of interactivity, in a sea of otherwise static content

But lets say I want two islands to have the same state. How do I make them communicate with each other?

**Comment by lucacasonato** [comment]
Pass events between them using `window`?

**Reply by italodeandra** [reply]
For now I'm using [valtio](https://github.com/pmndrs/valtio) for sharing the state and seems to work fine. But I'm still investigating and testing, not sure how well it will work on a real project in production.

**Reply by OldManFroggy** [reply]
> Pass events between them using `window`?

I did this, now am using a state store: [Statery](https://github.com/hmans/statery) which works a lot like a svlete store, and seems to work well even at moderate scale.

**Comment by vorcigernix** [comment]
https://preactjs.com/guide/v10/hooks#usereducer

**Comment by czottmann** [comment]
@italodeandra I did a write-up on this two days ago!  [How to use cross-island state management in Deno Fresh
 ](https://zottmann.org/2022/07/31/how-to-use.html)  TL;DR: use a state-management store.

Passing events or data around using `window` is prone to breaking, IME.  You "reach outside" the app, IMHO there's too much risk of other outside colliding.  Using a store and Preact/ React in a clean work is easier to maintain and test.

**Comment by marvinhagemeister** [comment, marked as answer]
FYI: We now have a section about that in our docs https://fresh.deno.dev/docs/examples/sharing-state-between-islands
